pilot/pkg/model/push_context.go

```diff
@@ -326,6 +457,17 @@ func (ps *PushContext) initServiceRegistry(env *Environment) error {
 	ps.Services = sortServicesByCreationTime(services)
 	for _, s := range services {
 		ps.ServiceByHostname[s.Hostname] = s
+
+		// Update service2ports
+		ports := map[string]uint32{}
+		portsByNum := map[uint32]string{}
+
+		for _, port := range s.Ports {
+			ports[port.Name] = uint32(port.Port)
+			portsByNum[uint32(port.Port)] = port.Name
+		}
+		ps.ServiceName2Ports[string(s.Hostname)] = ports
+		ps.ServicePort2Name[string(s.Hostname)] = portsByNum
```

Comment:
isn't this wasteful to have 2 maps when most of the services will be single port services?

Reply:
Good point - may be worth changing back. I see master removed the 'by name' - so maybe when I merge to master I remove that index. Size for map is not so bad, and we expect ~5k..10k services, so not a huge issue..